addons/dns/enable

```diff
@@ -6,53 +6,61 @@ source $SNAP/actions/common/utils.sh
 CURRENT_DIR=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
 source $CURRENT_DIR/../common/utils.sh
 
-KUBECTL="$SNAP/kubectl --kubeconfig=${SNAP_DATA}/credentials/client.config"
+KUBECTL="$SNAP/microk8s-kubectl.wrapper"
+
 # Apply the dns yaml
 # We do not need to see dns pods running at this point just give some slack
 echo "Enabling DNS"
 
 read -ra ARGUMENTS <<< "$1"
-if [[ ! -z "${ARGUMENTS[@]}" ]]
-then
-  nameservers="${ARGUMENTS[@]}"
-else
-  nameservers="8.8.8.8,8.8.4.4"
-fi
+if [[ -z "${ARGUMENTS[@]}" ]] || [[ "${ARGS[@]}" == "/etc/resolv.conf" ]]; then
+  resolv_conf="$(cat $SNAP_DATA/args/kubelet | grep -- "--resolv-conf" | tr "=" " " | gawk '{print $2}')"
+  if [ ! -z "${resolv_conf}" ]; then
+    # kubelet has a resolv.conf file configured, we will use that one
+    echo "Using host configuration from ${resolv_conf}"
+    nameserver_str="/etc/resolv.conf"
+  else
+    host_resolv_conf="$($SNAP/usr/bin/python3 $SNAP/scripts/find-resolv-conf.py 2> /dev/null || true)"
```

Comment:
This is a bit dangerous. Would `$SNAP/scripts/find-resolv-conf.py` be present in a 1.25 microk8s? Could/should we guard against this?